### PR TITLE
Listings

### LINKED ISSUE #25: getListingStatus
This request is made from the New Offer page before the post request is made (in case the listing has been accepted in the time the user created an offer) 
- [x] query that responds with listing id and its status

### LINKED ISSUE #26: deleteOffer
- [ ] deleteOffer

### PR DESCRIPTION
**Pull request by jdcarey128**
## Description

This PR completes the query for getListingStatus and the mutation for deleteOffer.
closes #26 
closes #25 

## Motivation and Context

Frontend thought it would be an easy win to implement a delete for offers but not necessarily the update functionality. We can add this later if they want us to. 

## How Has This Been Tested?

Normal and edge-case testing. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/graphql/mutations/delete_offer.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Mutations
```
Comment (Callbritton) [inline app/graphql/mutations/delete_offer.rb:1]: This will be really useful!

**Review by Callbritton — APPROVED** [review]
Looks great!

File: spec/requests/mutations/delete_offer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    result = JSON.parse(response.body, symbolize_names: true)
+    deleted_offer = result[:data][:deleteOffer][:offer]
+    error = result[:data][:deleteOffer][:error]
+
+    expect(error).to be_empty
+    expect(deleted_offer[:id]).to eq(offer.id)
+    expect(Offer.find_by(id: offer.id)).to eq(nil)
+  end
+
+  it 'unless the offer id is invalid' do
```
Comment (dcoleman21) [inline spec/requests/mutations/delete_offer_spec.rb:30]: Nice edge case testing here

**Review by dcoleman21 — APPROVED** [review]
This looks great Joshua.

**Pull request merged**